modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -446,15 +447,26 @@ def mask(
         If both `row_indices` and `row_numeric_idx` are set, `row_indices` will be used.
         The same rule applied to `col_indices` and `col_numeric_idx`.
         """
-        # Check on all possible ranges
```

Comment:
@dchigarev the analogous method in cuDFOnRayDataframe still uses the old code here.  any reason not to update there too?

Reply:
It seems that I've just missed `cuDFOnRayDataframe` when I did this PR, so basically, there's no reason not to propagate the same logic to the [`cuDFOnRayDataframe.take_2d_labels_or_positional`](https://github.com/modin-project/modin/blob/dff84b9cdf373096f88356fe17202599c192c2fc/modin/core/execution/ray/implementations/cudf_on_ray/dataframe/dataframe.py#L120)